BrainUp/BrainUp/Network/NetworkService.swift

```diff
@@ -0,0 +1,12 @@
+//
+//  NetworkService.swift
+//  BrainUp
+//
+//  Created by Evgenii Zhigunov on 2/18/22.
+//
+
+import Foundation
+
+protocol NetworkService {
+    func fetch<T: Codable>(_ request: Request, completion: @escaping (Result<T, Error>) -> Void)
```

Comment:
Should we use here Codable? I suggest use only Decodable.

Reply:
fixed